undercrawler/spiders/base_spider.py

```diff
@@ -362,3 +357,25 @@ def link_to_url(link):
     if link.fragment and link.fragment != '#':
         return "#".join([link.url, link.fragment])
     return link.url
+
+
+def allowed_re(url, hard_url_constraint):
```

Comment:
@kmike I've updated the regexp to allow subdomains: it's a little bit more permissive than what is allowed by domain spec, but I believe there are not valid urls that could be handled incorrectly.